flang/lib/Lower/CharacterRuntime.cpp

```diff
@@ -131,6 +131,24 @@ void Fortran::lower::genIndexDescriptor(Fortran::lower::FirOpBuilder &builder,
                      substringBox, backOpt, kind);
 }
 
+void Fortran::lower::genRepeat(Fortran::lower::FirOpBuilder &builder,
+                             mlir::Location loc, mlir::Value resultBox,
+                             mlir::Value stringBox, mlir::Value ncopies) {
+  auto repeatFunc = getRuntimeFunc<mkRTKey(Repeat)>(loc, builder);
+  auto fTy = repeatFunc.getType();
+  auto sourceFile = Fortran::lower::locationToFilename(builder, loc);
+  auto sourceLine =
+      Fortran::lower::locationToLineNo(builder, loc, fTy.getInput(4));
+
+  llvm::SmallVector<mlir::Value, 5> args;
+  args.emplace_back(builder.createConvert(loc, fTy.getInput(0), resultBox));
+  args.emplace_back(builder.createConvert(loc, fTy.getInput(1), stringBox));
+  args.emplace_back(builder.createConvert(loc, fTy.getInput(2), ncopies));
+  args.emplace_back(builder.createConvert(loc, fTy.getInput(3), sourceFile));
+  args.emplace_back(builder.createConvert(loc, fTy.getInput(4), sourceLine));
```

Comment:
The tutorial needs a small update here, you can now do
`auto args = Fortran::lower::createArguments(builder, loc, fTy, [the args in order....])`  to avoid manually doing the type cast.

Reply:
Using createArguments now.